lib/puppetserver/ca/action/prune.rb

```diff
@@ -80,7 +80,11 @@ def prune_CRLs(crl_list)
 
         def update_pruned_CRL(crl_list, pkey)
           crl_list.each do |crl|
-            crl.version=(crl.version + 1)
+            crl.extensions.each do |ext|
```

Comment:
You should be able to use `select` rather than iterating over each extension.

Reply:
Not to get into golfing territory but [partition](https://ruby-doc.org/core-2.5.1/Enumerable.html#method-i-partition) is what you want.

```
irb(main):001:0> a = [ 1, 2, 3, 4, 5, 6]
irb(main):003:0> even, odd = a.partition(&:even?)
irb(main):004:0> even
=> [2, 4, 6]
irb(main):005:0> odd
=> [1, 3, 5]
```